src/main/java/com/google/firebase/remoteconfig/ConditionEvaluator.java

```diff
@@ -179,6 +185,78 @@ private boolean evaluateCustomSignalCondition(CustomSignalCondition condition,
     }
   }
 
+  private boolean evaluatePercentCondition(PercentCondition condition,
+      KeysAndValues context) {
+    if (!context.containsKey("randomizationId")) {
+      logger.warn("Percentage operation must not be performed without randomizationId");
+      return false;
+    }
+
+    PercentConditionOperator operator = condition.getPercentConditionOperator();
+
+    // The micro-percent interval to be used with the BETWEEN operator.
+    MicroPercentRange microPercentRange = condition.getMicroPercentRange();
+    int microPercentUpperBound = microPercentRange != null
+        ? microPercentRange.getMicroPercentUpperBound()
+        : 0;
+    int microPercentLowerBound = microPercentRange != null
+        ? microPercentRange.getMicroPercentLowerBound()
+        : 0;
+    // The limit of percentiles to target in micro-percents when using the
+    // LESS_OR_EQUAL and GREATER_THAN operators. The value must be in the range [0
+    // and 100000000].
+    int microPercent = condition.getMicroPercent();
+    BigInteger microPercentile = getMicroPercentile(condition.getSeed(),
+        context.get("randomizationId"));
+    switch (operator) {
+      case LESS_OR_EQUAL:
+        return microPercentile.compareTo(BigInteger.valueOf(microPercent)) <= 0;
+      case GREATER_THAN:
+        return microPercentile.compareTo(BigInteger.valueOf(microPercent)) > 0;
+      case BETWEEN:
+        return microPercentile.compareTo(BigInteger.valueOf(microPercentLowerBound)) > 0
+            && microPercentile.compareTo(BigInteger.valueOf(microPercentUpperBound)) <= 0;
+      case UNSPECIFIED:
+      default:
+        return false;
+    }
+  }
+
+  private BigInteger getMicroPercentile(String seed, String randomizationId) {
+    String seedPrefix = seed != null && !seed.isEmpty() ? seed + "." : "";
+    String stringToHash = seedPrefix + randomizationId;
+    BigInteger hash = hashSeededRandomizationId(stringToHash);
+    BigInteger modValue = new BigInteger(Integer.toString(100 * 1_000_000));
```

Comment:
nit: would it make sense to define this in a const?
`private static final BigInteger MICRO_PERCENT_MODULO = BigInteger.valueOf(100_000_000L);`